go/arrow/array/float16.go

```diff
@@ -0,0 +1,85 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I think we should test the logic of converting to/from float32 to float16 even though it seems to be partially tested by the buidler

Reply:
will do